File: Tema5_2/src/main/kotlin/classes/Institut.java
package classes;

import javax.persistence.*;
import java.util.Objects;

@Entity
public class Institut {
    private String codi;
    private String nom;
    private String adreca;
    private String numero;
    private Integer codpostal;
    private Poblacio poblacioByCodM;

    @Id
    @Column(name = "codi")
    public String getCodi() {
        return codi;
    }

    public void setCodi(String codi) {
        this.codi = codi;
    }

    @Basic
    @Column(name = "nom")
    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    @Basic
    @Column(name = "adreca")
    public String getAdreca() {
        return adreca;
    }

    public void setAdreca(String adreca) {
        this.adreca = adreca;
    }

    @Basic
    @Column(name = "numero")
    public String getNumero() {
        return numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    @Basic
    @Column(name = "codpostal")
    public Integer getCodpostal() {
        return codpostal;
    }

    public void setCodpostal(Integer codpostal) {
        this.codpostal = codpostal;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Institut institut = (Institut) o;
        return Objects.equals(codi, institut.codi) && Objects.equals(nom, institut.nom) && Objects.equals(adreca, institut.adreca) && Objects.equals(numero, institut.numero) && Objects.equals(codpostal, institut.codpostal);
    }

    @Override
    public int hashCode() {
        return Objects.hash(codi, nom, adreca, numero, codpostal);
    }

    @ManyToOne
    @JoinColumn(name = "cod_m", referencedColumnName = "cod_m")
    public Poblacio getPoblacioByCodM() {
        return poblacioByCodM;
    }

    public void setPoblacioByCodM(Poblacio poblacioByCodM) {
        this.poblacioByCodM = poblacioByCodM;
    }
}

File: Tema5_PostgreSQL_geo_ad/src/main/kotlin/classes/Institut.kt
package classes

import classes.Poblacio
import classes.Institut
import javax.persistence.*
import java.io.Serializable
@Entity
class Institut:Serializable {
    companion object {
        private const val serialVersionUID: Long = 1
    }

    @get:Column(name = "codi")
    @get:Id
    var codi: String? = null

    @get:Column(name = "nom")
    @get:Basic
    var nom: String? = null

    @get:Column(name = "adreca")
    @get:Basic
    var adreca: String? = null

    @get:Column(name = "numero")
    @get:Basic
    var numero: String? = null

    @get:Column(name = "codpostal")
    @get:Basic
    var codpostal: Int? = null

    @get:JoinColumn(name = "cod_m", referencedColumnName = "cod_m")
    @get:ManyToOne
    var poblacio: Poblacio? = null
    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val institut = o as Institut
        if (if (codi != null) codi != institut.codi else institut.codi != null) return false
        if (if (nom != null) nom != institut.nom else institut.nom != null) return false
        if (if (adreca != null) adreca != institut.adreca else institut.adreca != null) return false
        if (if (numero != null) numero != institut.numero else institut.numero != null) return false
        return if (if (codpostal != null) codpostal != institut.codpostal else institut.codpostal != null) false else true
    }

    override fun hashCode(): Int {
        var result = if (codi != null) codi.hashCode() else 0
        result = 31 * result + if (nom != null) nom.hashCode() else 0
        result = 31 * result + if (adreca != null) adreca.hashCode() else 0
        result = 31 * result + if (numero != null) numero.hashCode() else 0
        result = 31 * result + if (codpostal != null) codpostal.hashCode() else 0
        return result
    }
}